Modules/_struct.c

```diff
@@ -1930,15 +1930,40 @@ s_pack_into(PyObject *self, PyObject **args, Py_ssize_t nargs, PyObject *kwnames
         return NULL;
     }
 
+    /* Check that negative offset is low enough to fit data */
+    if (offset < 0 && offset + soself->s_size > 0) {
+        PyErr_Format(StructError,
+                     "pack_into requires negative offset not higher than %zd "
```

Comment:
This just repeats the same number two times (but with different signs). It would be more helpful to output an actual value of `offset`.